### PR TITLE
release: setup logging when ensuring operator installed

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/4355.

### CODE REVIEW

File: pkg/release/release.go [hunk 1]
```diff
@@ -184,10 +185,14 @@ func (r *Release) EnsureDeleted(ctx context.Context, name string, conditions ...
...
 
 // EnsureInstalled makes sure the release is installed and all conditions are
-// met.
+// met. If release name ends with "-operator" suffix it also selects
+// a "app=${name}" pod and streams it logs to the ./logs directory.
 //
 // NOTE: It does not update the release if it already exists.
 func (r *Release) EnsureInstalled(ctx context.Context, name string, version Version, values string, conditions ...func() error) error {
+	var err error
+	isOperator := strings.HasSuffix(name, "-operator")
```
Comment (kopiczko) [inline pkg/release/release.go]: Not sure if this is the best approach on earth. But previous one was also based on heavy assumptions.

Comment (MarcelMue) [reply pkg/release/release.go]: What is the benefit of doing this insteal of `InstallOperator` and using `EnsureInstalled` inside?

Comment (MarcelMue) [reply pkg/release/release.go]: To me this does not seem intuitive, when I call `InstallOperator` I expect something different to happen than just calling `Install`. Here there is no indication that something different is happening behind the scenes.

Comment (kopiczko) [reply pkg/release/release.go]: I want to have idempotent version of installing an operator. Also having consistency is nice. What's the alternative? `EnsureOperatorInstalled`? 

To make the logging nice I would write exactly like that:

```
func (r *Release) EnsureInstalled(ctx context.Context, name string, version Version, values string, conditions ...func() error) error {
	err := r.ensureInstalled(ctx, name, version, values, false, conditions...)
	if err != nil {
		return microerror.Mask(err)
	}
	
	return nil
}

func (r *Release) EnsureOperatorInstalled(ctx context.Context, name string, version Version, values string, conditions ...func() error) error {
	err := r.ensureInstalled(ctx, name, version, values, true, conditions...)
	if err != nil {
		return microerror.Mask(err)
	}
	
	return nil
}

func (r *Release) ensureInstalled(ctx context.Context, name string, version Version, values string, isOperator bool, conditions ...func() error) error {
}
```

Is that better?

Comment (MarcelMue) [reply pkg/release/release.go]: I don't get why you want to mix both things in `ensureInstalled`. `EnsureOperatorInstalled` adds additional functionality around the base functionality of `EnsureInstalled`. 

May solution would be to simply replace this line here with `EnsureInstalled`:
https://github.com/giantswarm/e2e-harness/blob/9f97abe68c8fb2ce7a50886098d4d4d24d187583/pkg/release/release.go#L283

Comment (kopiczko) [reply pkg/release/release.go]: I'd like to setup logging before we start waiting for CRD. Otherwise we can miss some error.

Comment (kopiczko) [reply pkg/release/release.go]: I mean error during CRD creation in the operator. That was a flaw of previous implementation.

Comment (MarcelMue) [reply pkg/release/release.go]: Okay I see your point. I still would not want two functions doing the same thing in different ways (`InstallOperator` and `EnsureInstalled`). So if that is a problem then let's fix it for everything.

File: pkg/release/release.go [hunk 2]
```diff
@@ -201,6 +206,38 @@ func (r *Release) EnsureInstalled(ctx context.Context, name string, version Vers
...
+		err := r.waitForConditions(ctx, c)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("ensured operator pod exists for release %#q", name))
+	}
+
+	var operatorPodName string
+	var operatorPodNamespace string = r.namespace
```
Comment (MarcelMue) [inline pkg/release/release.go]: I think you might need to same fallback as `InstallOperator` to the GS namespace. I am not sure if the namespacing is right everywhere currently.

File: pkg/release/release.go (hunk 1 above)
Comment (kopiczko) [reply pkg/release/release.go]: I'd like to get rid of `InstallOperator` completely.
I'd also like to remove `conditions` form `Install` and `Upadate` (where they aren't used btw) and make those methods lean. We should use only `Ensure*` implementations unless we have very special cases. In such cases I think we can use conditions explicitly.

File: pkg/release/release.go (hunk 2 above)
Comment (kopiczko) [reply pkg/release/release.go]: Yes. I thought it's a good opportunity to test if it is right.

File: pkg/release/release.go (hunk 1 above)
Comment (kopiczko) [reply pkg/release/release.go]: > I'd like to get rid of InstallOperator completely.

Maybe it's a good idea to do this in this PR?

Comment (MarcelMue) [reply pkg/release/release.go]: I agree, let's remove `InstallOperator` in this PR then and vendor this into our major operators. If everything goes well, then we can merge here.

File: pkg/release/release.go [hunk 3]
```diff
@@ -242,6 +279,10 @@ func (r *Release) Install(ctx context.Context, name string, version Version, val
 	return nil
 }
 
+// TODO Remove once we are done with migration to EnsureInstalled.
+//
+//	Issue https://github.com/giantswarm/giantswarm/issues/4355.
+//
```
Comment (kopiczko) [inline pkg/release/release.go]: @MarcelMue I added TODO instead. This is the same as approach in `framework/resource` in general.

File: pkg/release/release.go (hunk 1 above)
Comment (kopiczko) [reply pkg/release/release.go]: https://github.com/giantswarm/e2e-harness/pull/155#discussion_r228425284

**Comment by kopiczko** [comment]
@MarcelMue PTALA

**Review by MarcelMue — APPROVED** [review]
Let's go with this, the follow up is mentioned in the issue so this is fine.

**Pull request merged**